Hearthstone Deck Tracker/Windows/DeckScreenshotWindow.xaml.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using Hearthstone_Deck_Tracker.Hearthstone;
+using Hearthstone_Deck_Tracker.Utility.Extensions;
+using static System.Windows.Visibility;
+
+namespace Hearthstone_Deck_Tracker.Windows
+{
+	public partial class DeckScreenshotWindow
+	{
+		private const int EXTRAS_HEIGHT = 40 + 24 + 34 + 24;
+		private string _allTags;
+
+		public DeckScreenshotWindow(Deck deck, bool deckOnly = false)
+		{
+			InitializeComponent();
+			_allTags = deck.TagList.ToLowerInvariant().Replace("-", "");
+			ListViewPlayer.Update(deck.Cards.ToSortedCardList(), true);
+
+			if(deckOnly)
+			{
+				DeckTitleContainer.Visibility = Collapsed;
+				DeckFormatPanel.Visibility = Collapsed;
+				SetDustPanel.Visibility = Collapsed;
+				BrandContainer.Visibility = Collapsed;
+			}
+			else
+			{
+				DeckTitlePanel.Background = DeckHeaderBackground(deck.Class);
+				LblDeckTitle.Text = deck.Name;
+				LblDeckTag.Text = GetTagText(deck);
+				LblDeckFormat.Text = GetFormatText(deck);
+				LblDustCost.Text = TotalDust(deck).ToString();
+				ShowFormatIcon(deck);
+				SetIcons.Update(deck);
+			}
+
+			Height = 35 * ListViewPlayer.Items.Count + (deckOnly ? 0 : EXTRAS_HEIGHT);
+		}
+
+		public double DeckMaxHeight => ActualHeight - LabelsHeight;
+
+		public double LabelsHeight => DeckTitleContainer.ActualHeight + DeckFormatPanel.ActualHeight
+			+ SetDustPanel.ActualHeight + BrandContainer.ActualHeight + 42;
+
+		public void UpdateWindowHeight() => Height = StackPanelMain.ActualHeight + 42;
+
+		private ImageBrush DeckHeaderBackground(string deckClass)
+		{
+			var heroId = ClassToID(deckClass);
+			var drawingGroup = new DrawingGroup();
+			drawingGroup.Children.Add(new ImageDrawing(new BitmapImage(new Uri(
+				$"Images/Bars/{heroId}.png", UriKind.Relative)), new Rect(54, 0, 130, 34)));
+			drawingGroup.Children.Add(new ImageDrawing(new BitmapImage(new Uri(
+				"Images/Themes/Bars/dark/fade.png", UriKind.Relative)), new Rect(0, 0, 183, 34)));
+
+			return new ImageBrush {
+				ImageSource = new DrawingImage(drawingGroup),
+				AlignmentX = AlignmentX.Left,
+				Stretch = Stretch.UniformToFill
+			};
+		}
+
+		private string GetTagText(Deck deck)
+		{
+			var predefined = new List<string>() {
+				"Midrange",
+				"Aggro",
+				"Control",
+				"Tempo",
+				"Combo"
+			};
+
+			if(deck.Tags.Count > 0)
+				foreach(var tag in predefined)
+					if(_allTags.Contains(tag.ToLowerInvariant()))
+						return tag;
+
+			return deck.Class;
+		}
+
+		private string GetFormatText(Deck deck)
+		{
+			if(deck.IsArenaDeck)
+				return "Arena";
+			else if(_allTags.Contains("brawl"))

```

Comment:
Sorry, I meant all of the 'else' here are unnecessary, not just the last one :).